### PR TITLE
Add date types

### PR DESCRIPTION
**Pull request by alyonavyshnevska**
Added four new types:
- `CdsDate`
- `CdsTme`
- `CdsDateTime`
- `CdsTimeStamp`

which correspond to the built-in types `Date`, `Time`, `DateTime` and `Timestamp` in `cds`. 

During runtime, these types are just strings. `CdsDate` and `CdsTime` each come with a regex that only accepts values in the same format as `cds.Date` and `cds.Time`. 
 
resolves https://github.com/cap-js/cds-typer/issues/164

- [x] changelog

### CODE REVIEW

File: lib/file.js [hunk 1]
```diff
@@ -634,6 +637,15 @@ export type EntitySet<T> = T[] & {
...
 } & Exclude<Required<T>, null>;
+
+
+/**
+ * Dates and timestamps are strings during runtime, so cds-typer represents them as such.
+ */
+export type CdsDate = ${dateRegex};
+export type CdsDateTime = string;
+export type CdsTime = ${timeRegex};
+export type CdsTimestamp = string;
```
Comment (daogrady) [inline lib/file.js:648]:
```suggestion
export type CdsTimeStamp = string;
```

**Review by daogrady — CHANGES_REQUESTED** [review]
Looks mostly good, just the spelling on the generated type is slightly out of line with how it is generated in the builtins (which leads to an "unresolved type" error in the generated files). See my comments.

File: lib/file.js (hunk 1 above)
Comment (alyonavyshnevska) [reply lib/file.js:648]: ah, actually this should be the other way around to be consistent with the [built-in types](https://pages.github.tools.sap/cap/docs/guides/domain-modeling#standard-built-in-types). Good that you noticed the discrepancy there.

File: lib/file.js [hunk 2]
```diff
@@ -634,6 +637,15 @@ export type EntitySet<T> = T[] & {
 export type DeepRequired<T> = { 
     [K in keyof T]: DeepRequired<T[K]>
 } & Exclude<Required<T>, null>;
+
+
+/**
+ * Dates and timestamps are strings during runtime, so cds-typer represents them as such.
+ */
+export type CdsDate = ${dateRegex};
```
[the remark this answers is not in the document]
Comment (alyonavyshnevska) [reply lib/file.js:645]: My mistake, did not commit the change

**Review by daogrady — APPROVED** [review]
Looks good now, thanks for the contribution! :)

**Pull request merged**